src/components/Card.js

```diff
@@ -1,53 +1,98 @@
+    this._owner = data.owner._id;
+    this._handleDeleteCard = handleDeleteCard;
+    console.log(this._owner);
   }
 
-  _setEventListeners() {
-    this._cardLikeButton.addEventListener("click", () => {
-      this._handleLikeIcon();
-    });
+  _getTemplate() {
+    const cardElement = document
+      .querySelector(this._cardSelector)
+      .content.firstElementChild.cloneNode(true);
+    return cardElement;
+  }
 
-    this._cardDeleteButton.addEventListener("click", () => {
-      this._handleDeleteCard();
-    });
+  deleteClick() {
+    this._element.remove();
+  }
 
-    this._cardImage.addEventListener("click", () => {
-      this._handleImageClick({ name: this._name, link: this._link });
-    });
+  updateLikes(isLiked) {
+    this._isLiked = isLiked;
+    this.renderLikes();
   }
 
-  _handleLikeIcon() {
-    this._cardLikeButton.classList.toggle("card__like-button_active");
+  renderLikes() {
+    // this._cardLikes.textContent = this._likes.length;
+    const isLiked = this.isLiked();
+    if (isLiked) {
+      this.likeButton.classList.add("card__like-button_active");
+    } else {
+      this.likeButton.classList.remove("card__like-button_active");
+    }
   }
 
-  _handleDeleteCard() {
-    this._cardElement.remove();
+  isLiked() {
+    return this._isLiked;
   }
 
-  _getTemplate() {
-    return document
-      .querySelector(this._cardSelector)
-      .content.querySelector(".card")
-      .cloneNode(true);
+  _checkIdForDelete() {
+    if (this._owner === this._currentUserId) {
+      this.addTrash();
+    } else {
+      this.removeTrash();
+    }
   }
 
-  getView() {
-    this._cardElement = this._getTemplate();
-    this._cardLikeButton =
-      this._cardElement.querySelector(".card__like-button");
-    this._cardDeleteButton = this._cardElement.querySelector(
-      ".card__delete-button"
+  addTrash() {
+    this.cardTrashButton.classList.remove("card__trash-button_active");
+  }
+
+  removeTrash() {
+    this.cardTrashButton.classList.add("card__trash-button_active");
+  }
+
+  _setEventListeners() {
+    this.likeButton.addEventListener("click", () =>
+      this._handleLikeCard(this.isLiked())
     );
-    this._cardImage = this._cardElement.querySelector(".card__image");
-    this._cardImage.src = this._link;
-    this._cardImage.alt = this._name;
-    this._cardElement.querySelector(".card__title").textContent = this._name;
-    this._setEventListeners();
+    this.cardTrashButton.addEventListener("click", () => {
+      this._handleDeleteCard(this._id);
+    });
+    this.cardImage.addEventListener("click", () => {
+      this._handleImageClick();
+    });
+  }
 
-    return this._cardElement;
+  generateCard() {
+    this._element = this._getTemplate();
+    this.likeButton = this._element.querySelector(".card__like-button");
+    this.cardTrashButton = this._element.querySelector(".card__trash-button");
+    this.cardImage = this._element.querySelector(".card__image");
```

Comment:
Good job that you've found these variables only once here and defined them as **class fields**